src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -13,21 +13,162 @@
 #
 #
 
-.rs.addJsonRpcHandler("preview_data_import", function(path, maxRows, maxCols = 100, maxFactors = 64)
+.rs.addFunction("assemble_data_import_name", function(dataImportOptions)
+{
+  dataName <- dataImportOptions$dataName
+
+  if (is.null(dataName) || dataName == "")
+    {
+      if (!is.null(dataImportOptions$importLocation) && dataImportOptions$importLocation != "")
+      {
+        basename <- basename(dataImportOptions$importLocation)
```

Comment:
It's not usually wise to shadow a function name with a local variable name (from now on `basename(foo)` will cause errors)